doc/sfp-cmis/Interface-Link-bring-up-sequence.md

```diff
@@ -0,0 +1,136 @@
+
+# Background
+
+  Per the ‘QSFPDD spec’,  ‘validation, diagnostics’ done by HW team' and 'agreement with vendors', 
+  need to follow following bring-up seq to enable port/interface with 400G optics in LC/chassis:
+
+    a) Enable port on NPU (bring-up port, serdes on the NPU ; enable signals) : syncd
+    b) Enable port on PHY (bring-up port, serdes on the PHY ; enable signals) : gbsyncd
+       - Wait for signal to stabilize on PHY   
+    c) Enable optical module (turn laser on/ enable tx) : xcvrd or platform bootstrap/infra
+
+  In boards not having PHY, #b) not needed but #a) and #c) sequence to be followed.
+  
+  ## Clause from QSFP-DD (CMIS4.0 spec)
+  
+     Excerpt from CMIS4.0 spec providing detailed reasoning for the above-mentioned bring-up sequence
+     
+  ![61f5b485-cf3b-4ca8-beac-9102b6feabfe](https://user-images.githubusercontent.com/69485234/147173702-f124fc9d-ef27-4816-b1a1-b4a44a5833a7.PNG)
+
+
+  ## Clause from QSFP-DD (CMIS5.0 spec)
+  
+     Excerpt from CMIS5.0 spec providing detailed reasoning for the above-mentioned bring-up sequence
+     
+  ![96a35dc5-618f-418c-9593-5639a90f1b28](https://user-images.githubusercontent.com/69485234/147173164-5ad0123c-479a-4774-b3ee-12a81fdd7d7e.PNG)
+     
+
+# Objective
+
+Have a determistic approach for Interface link bring-up sequence i.e. below sequence to be followed:
+  1. Initialize and enable NPU Tx and Rx path
+  2. For system with 'External' PHY: Initialize and enable PHY Tx and Rx on both line and host sides; ensure host side link is up 
+  3. Then only perform optics data path initialization/activation/Tx enable (for 400G) and Tx enable (for 100G) 
+
+# Proposal
+
+Recommend following this high-level work-flow sequence to accomplish the Objective:
+- xcvrd to subscribe to a new field “host_tx_ready” in port table state-DB
+- Orchagent will set the “host_tx_ready” to true/false based on the SET_ADMIN_STATE attribute return status to syncd/gbsyncd. (As part of SET_ADMIN_STATE attribute enable, the NPU Tx is enabled)
+- xcvrd process the “host_tx_ready” value change event and do optics datapath init / de-init using CMIS API
+- Recommendation is to follow this proposal for all the known interface speeds - 400G/100G/40G/25G/10G
+  - 400G - as mentioned above the CMIS spec to be followed
+  - 100G/40G/25G/10G - deterministic approach to bring the interface will eliminate any link stability issue which will be difficult to chase in the production network
+  - This synchronization will also benefit native 10G SFPs interfaces as they are "plug N play" and may not have quiescent functionality. (xcvrd can use the optional 'soft tx disable' ctrl reg to disable the tx)
+
+# Proposed Work-Flows
+Please refer to the  flow/sequence diagrams which covers the following required use-cases
+  - Transceiver initialization
+  - admin enable configurations 
+  - admin disable configurations
+
+# Transceiver Initialization 
```

Comment:
I believe we are moving towards a model where only xcvrd interacts with cables/transceivers and so I think that the platform_bootstrap steps would need to occur after xcvrd spawns.

Reply:
No. 
platform bootstrap comes first, even before, systemd spawns PMON, xcvrd etc.
sfp refactoring design expects the platform/vendor to detect the optics presence on the board/LC and take them out-of-reset and powers them on. So, all this expected to be done as part of platform bootstrap layer.